server/routes/appointment.js

```diff
@@ -0,0 +1,23 @@
+// routes/appointment.js
+const express = require('express');
+const router = express.Router();
+const sendEmail = require('../utils/sendEmail'); // Ensure the path is correct
+
+router.post('/', async (req, res) => {
+  const { name, email, date, time } = req.body;
+
+  try {
+    const subject = `Appointment Confirmation for ${name}`;
+    const text = `Your appointment is confirmed for ${date} at ${time}.`;
+
+    await sendEmail('wsucareconnect23@gmail.com', subject, text); // Send email to clinic
+    await sendEmail(email, subject, `Dear ${name},\n${text}`); // Send confirmation to user
+
+    res.status(200).json({ message: 'Appointment booked and email sent.' });
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ message: 'Error in booking appointment.' });
+  }
+});
+
+module.exports = router;
```

Comment:
The appointment booking feature on the website encounters functionality issues with email confirmation delivery. Users are not receiving their appointment confirmations, indicating a potential problem with the backend code or email configuration. Enhanced error handling and logging are recommended to identify and address issues promptly.